rewrite-java-test/src/test/java/org/openrewrite/java/JavaParserTest.java

```diff
@@ -68,6 +68,30 @@ public class PersistenceManagerImpl {
         );
     }
 
+    @Issue("https://github.com/openrewrite/rewrite/issues/3453")
+    @Test
+    void annotationBetweenDimensions() {
+        rewriteRun(
+          java(
+            """
+              import javax.annotation.Nonnull;
+
+              class ArrayNotNull {
+                  int[][] ints = new int[2][2];
+                  
+                  Integer[] @Nonnull [] integers = new Integer[1][1];
+                  
+                  Integer @Nonnull [] foo = new Integer[1];
+
+                  public int   @Nonnull    [] @Nonnull [] getInts() {
+                      return ints;
+                  }
```

Comment:
Thanks a lot for being thorough with the various forms of annotations and whitespace here! Good to have this covered.